### PR TITLE
Fix build system error introduced in #8596

### PR DESCRIPTION
**Pull request by cstrahan**
Description: Resolves the build system error introduced in #8596
Risk Level: Low
Testing: Built the proxy (`bazel build //source/exe:envoy-static`)
Fixes: #9987

### CODE REVIEW

**Comment by cstrahan** [comment]
I'll note that I left the source files split up for organization purposes -- the only thing that was reverted was the erroneous build system change.

File: source/server/BUILD [hunk 1]
```diff
@@ -352,23 +288,32 @@ envoy_cc_library(
...
         "//source/common/network:socket_option_factory_lib",
         "//source/common/network:utility_lib",
         "//source/common/protobuf:utility_lib",
+        "//source/common/stream_info:stream_info_lib",
         "//source/extensions/filters/listener:well_known_names",
         "//source/extensions/filters/network/http_connection_manager:config",
         "//source/extensions/transport_sockets:well_known_names",
         "@envoy_api//envoy/admin/v3:pkg_cc_proto",
+        "@envoy_api//envoy/api/v2/listener:pkg_cc_proto",
+        "@envoy_api//envoy/api/v2:pkg_cc_proto",
```
Comment (wrowe) [inline source/server/BUILD]: v2:pkg... sorts before v2/listen... (colon delimited first.) Switch these 2 to get linux-format to pass.

Comment (cstrahan) [reply source/server/BUILD]: Thanks - done.

**Comment by lizan** [comment]
@junr03 do you mind take a look? Seems related to this TODO https://github.com/envoyproxy/envoy/blob/master/source/server/BUILD#L335

**Comment by zuercher** [comment]
@cstrahan you'll need to add the DCO signoff to your commits to get that one test to pass.

**Comment by cstrahan** [comment]
@zuercher Done.

File: source/server/BUILD [hunk 2]
```diff
@@ -271,73 +270,10 @@ envoy_cc_library(
...
-        "//source/common/network:utility_lib",
-        "//source/common/protobuf:utility_lib",
-        "//source/extensions/filters/listener:well_known_names",
-        "//source/extensions/transport_sockets:well_known_names",
-        "@envoy_api//envoy/config/core/v3:pkg_cc_proto",
-        "@envoy_api//envoy/config/listener/v3:pkg_cc_proto",
-    ],
-)
-
-# TODO(junr03): actually separate this lib from the listener and api listener lib.
```
Comment (junr03) [inline source/server/BUILD]: Can you preserve this comment so that I can come back to this?

Comment (cstrahan) [reply source/server/BUILD]: Whoops -- definitely.

Comment (cstrahan) [reply source/server/BUILD]: Done.

**Comment by cstrahan** [comment]
Could this be reopened, please?

> Thanks. Could you update the PR title? Other than that lgtm

@junr03 sure -- what would you like the PR title to be?

**Comment by cstrahan** [comment]
Can we reopen this please? And is there any way to disable the stale bot for this? Unless I'm mistaken, there is consensus that this fixes a bug in the build system.

I'm willing to make any requested changes to see this merged in.

**Comment by mattklein123** [comment]
Please merge master.

**Comment by lizan** [comment]
Can you also update the PR description to reflect what it does? it is no longer revert.

**Comment by cstrahan** [comment]
@lizan I've updated the PR title and touched up the commit message to explain the change.

@mattklein123 Merged master.

**Pull request merged**